### ISSUE TITLE
deeplexicon produces different demultiplexing results for GPU and CPU mode.

### ISSUE THREAD

**Issue by adalisan** [biocorecrg/BioNextflow#17]
For the RNA samples I am running, demultiplexing gives significantly different results. GPU mode results has a lot more unknown assignments and very small number of  bc3 barcoded reads. Meanwhile CPU mode  produces much more reasonable results.

**Comment by lucacozzuto** [comment]
Are you using cuda10?

**Comment by ADelgadoT** [comment]
Hi!

Could you tell us in which GPU and cuda version are you currently running MoP? Thanks!

**Comment by lpryszcz** [comment]
Hi @adalisan , [deeplexicon doesn't support never Nvidia architectures](https://github.com/Psy-Fer/deeplexicon/issues/25). 
Unfortunately the MoP team can't solve this.

**Comment by adalisan** [comment]
@lpryszcz oh...
You should definitely have big WARNING messages for this. Despite a lot of experience with GPUs, I definitely did not expect this. Failing to do inference is one thing, but giving wrong results was definitely not within my expectations.

**Comment by adalisan** [comment]
Yes, I think I am using cuda 11 with a variety of GPU architectures, some are older and some newer

**Comment by adalisan** [comment]
It may be possible to use CUDA 10 in our environment using bash modules. Would that work, with the same deeplexicon containers?

**Comment by lpryszcz** [comment]
@lucacozzuto I agree we should warn the users about the issue. 
@adalisan we were trying to warn from the level of the program, but we couldn't make it feasible. 
what's really odd, when you execute deeplexicon without Docker, it fails nicely. but inside docker it runs and produces wrong results. 

> It may be possible to use CUDA 10 in our environment using bash modules. Would that work, with the same deeplexicon containers?

adding CUDA10 won't help, because support for newer architectures comes only in CUDA11, while deeplexicon was built against CUDA10. so until deeplexicon is updated, we can't really provide support for newer Nvidia GPUs. 
On the bright side, we're working on our own demultiplexing tool. so stay tuned.

**Comment by enovoa** [comment]
Thanks @adalisan - we have now updated this info in the [DeePlexiCon](https://github.com/Psy-Fer/deeplexicon) README and added a WARNING as well as the requriement of CUDA v10. As @lpryszcz mentioned above, we will be releasing very soon (hopefully a couple of weeks) a new version of MasterOfPores (version 3) that contains a new demultiplexing algorithm that works with newer GPU architectures. Thanks!
